### ISSUE TITLE
Default text does not update after change made in app.js

### ISSUE THREAD

**Issue by TwoFistedJustice** [docker/getting-started#286]
[Edit]
__
The problem may have been a result of a step being omitted in the tutorial. See follow up comment.
__

I'm running Kubuntu 20.04
Instead of VSCode, I am using PhpStorm v2022.1.1

**Working from this page**
[https://docs.docker.com/get-started/03_updating_app/](https://docs.docker.com/get-started/03_updating_app/)

**Problem:**
The default text "No items yet! Add one above!" does not change in the browser after changing it in app.js

**I tried the following:**
- Change the text to "PURPLE ELEPHANTS!" to make sure I'm not confounding similar texts 
- Followed the CLI instructions by hand
- Copied and pasted the CLI code
- Followed the Docker Desktop instructions
- Cleared browser cache in Chromium
- Opened localhost:3000 in Firefox and Konqueror to make sure it wasn't Chromium.
- Added items and deleted all items to see if the default text would change
- Closed and re-opened all of my active windows
- Completely shut down my computer and restarted everything from scratch including qemu-kvm.service

**What I expected to happen:**
The default app text when no items are added should read "PURPLE ELEPHANTS!"

**What actually happened:**
The default app text when no items are added reads "No items yet! Add one above!"

![docker-tutorial](https://user-images.githubusercontent.com/22874887/179323911-7eec3048-7084-4679-9ddd-b2aa97102031.png)

**Comment by TwoFistedJustice** [comment]
I solved this after clearing my head by writing out everything I did above.

The issue is that the command
`$  docker build -t getting-started .`
must be run prior to running
`$  docker run -dp 3000:3000 getting-started`

I believe that step should be added to 
[#start-the-updated-app-container](https://docs.docker.com/get-started/03_updating_app/#start-the-updated-app-container)

Such that:
**Step 1:**
`$  docker build -t getting-started .`

**Step 2:**
$  docker run -dp 3000:3000 getting-started

**Step 3:**
Refresh your browser on [http://localhost:3000/](http://localhost:3000/) and you should see your updated help text!


I will leave the issue un-closed.

**Comment by mikesir87** [comment]
Thanks for the issue @TwoFistedJustice and apologies for the delay in getting back to you (we're promising to do better going forward with response times!).

Curious... were you using the getting started guide through the Docker Docs or through the write up found when running `docker run -d -p 80:80 docker/getting-started`? We have realized there is drift in the two docs and are working to fix that, but wanted to see if you can recall which guide you were using.

Beyond that... yes... we can do better with ensuring the steps are logical and help someone actually progress without little stumbling blocks like you brought up. And we're hoping to do exactly that now! So... stay tuned and definitely let us know if there were other issues you ran into while learning about Docker and/or containers. Thanks!

**Comment by mikesir87** [comment]
Since it's been just over a week, I'm going to go ahead and close this issue. Feel free to reopen/add additional comments if you have more discussion to share. Thanks!

**Comment by maxecharles** [comment]
Hi,

I'm working through the Docker tutorial and got stuck on exactly this. Definitely was a stumbling block that had me confused for a good hour trying to figure out why the app wasn't updating. I only figured it out when I found this (closed) issue.

Aware that this is 4 years down the track, but I agree with OP @TwoFistedJustice that the step should definitely be added to the tutorial. I was using the write up found when running `docker run -d -p 80:80 docker/getting-started`.

Have otherwise had good experience with tutorial thus far! Thanks for making it.

<img width="1000" height="812" alt="Image" src="https://github.com/user-attachments/assets/d4972ade-de60-45dd-945a-6def24c5ad6b" />
